cranelift/codegen/meta/src/isa/riscv64.rs

```diff
@@ -28,12 +28,23 @@ macro_rules! define_zvl_ext {
 pub(crate) fn define() -> TargetIsa {
     let mut setting = SettingGroupBuilder::new("riscv64");
 
-    let _has_m = setting.add_bool("has_m", "has extension M?", "", false);
-    let _has_a = setting.add_bool("has_a", "has extension A?", "", false);
-    let _has_f = setting.add_bool("has_f", "has extension F?", "", false);
-    let _has_d = setting.add_bool("has_d", "has extension D?", "", false);
+    // We target a minimum of riscv64gc. That means that we have the following extensions by default:
+    //
+    // * M (integer multiplication and division)
+    // * A (atomic instructions)
+    // * F (single-precision floating point)
+    // * D (double-precision floating point)
+    // * C (compressed instructions)
+    // * Zicsr (control and status register instructions)
+    // * Zifencei (instruction-fetch fence)
+
+    let _has_m = setting.add_bool("has_m", "has extension M?", "", true);
+    let _has_a = setting.add_bool("has_a", "has extension A?", "", true);
+    let _has_f = setting.add_bool("has_f", "has extension F?", "", true);
+    let _has_d = setting.add_bool("has_d", "has extension D?", "", true);
```

Comment:
Question that's somewhat unrelated to this PR but this inspired me to ask. IIRC I don't remember seeing lots of `has_f` checks, for example, in the `lower.isle` or `inst.isle` bits we have for RISC-V. I'm not sure whether that extension is itself relevant here, but this seems like a bug perhaps?

For example if `has_f` is disabled then I'd expect the backend would either automatically pick other instructions or if it's a require extension then we'd generate an error during construction of the ISA saying "required features are not enabled". Otherwise I think today what might happen is that if `has_f` is disabled then we'd generate instructions from the `f` instruction set?

(and again sorry I don't mean to pick on `f` here, I don't actually know what any of these extensions are, I just picked it randomly and this question applies to all of these enabled-by-default ones here)

Reply:
> Otherwise I think today what might happen is that if has_f is disabled then we'd generate instructions from the f instruction set?

Yeah that is exactly what happens. I don't think we ever check these base features in the backend only the non default ones. 

Adding those checks when constructing the ISA is a really good idea, I'm going to do that in a follow up PR.

The `f` extension is support for single precision floating point, the rest of them are equally basic stuff like that (I'm also going to add that to the flag description)

Reply:
Ok cool makes sense! If it helps this is the example I was thinking of that used to be in the x64 backend:

https://github.com/bytecodealliance/wasmtime/blob/66fd3d72c9d96b06df2dc5d930f12ccaa98f23f6/cranelift/codegen/src/isa/x64/mod.rs#L225-L233

that's since been removed for other reasons but I imagine there's a similar location to put this for the RISC-V backend.